src/plugins/send_events.mjs

```diff
@@ -30,12 +39,34 @@ const packagerConnect = async (rootPath) => {
         console.log('Error sending message to Sleeper App:', err);
       }
     });
+  });
 
-    client.on('error', () => {
-      client.destroy();
-    });
+  client.on('error', (error) => {
+    if (error?.code === 'ECONNREFUSED' && error?.syscall === 'connect') {
+      // We don't care about this error since we will retry the connection
+      return;
+    }
+
+    console.log('Socket Error: ', error);
+  });
+
+  client.on('data', (data) => {
+    // const json = JSON.parse(data);
+    // switch (json?.type) {
+    // }
   });
 
+  client.on('close', (hadError) => {
+    if (!hadError) {
+      console.log('Connection to Sleeper App closed, retrying...');
+    }
+
+    // Retry connection
+    setTimeout(socketConnect, refreshTimeout, client, appConfig);
```

Comment:
Is the reason this is a `setTimeout` and not `setInterval` because initiating `client.connect` will result in a `client.on('close')` if the connection is unsuccessful? (so the timeout gets triggered again?)

Reply:
This is correct. This close code is called whenever a connection fails to connect or is disconnected. setInterval could work as well with some re-worked logic, but this way is a bit more synchronous.